src/features/profile/pages/NotificationSettings/NotificationsSettings.native.test.tsx

```diff
@@ -255,14 +271,73 @@ describe('NotificationSettings', () => {
     ).not.toBeOnTheScreen()
   })
 
+  describe('When the user saves the changes', () => {
+    it('should update profile', async () => {
+      mockUseAuthContext.mockReturnValueOnce(baseAuthContext)
+      mockServer.postApi<UserProfileResponse>('/v1/profile', beneficiaryUser)
+
+      render(reactQueryProviderHOC(<NotificationsSettings />))
+
+      const toggleEmail = screen.getByTestId('Interrupteur Autoriser l’envoi d’e-mails')
+      await act(async () => fireEvent.press(toggleEmail))
+
+      const saveButton = screen.getByText('Enregistrer')
+      await act(async () => fireEvent.press(saveButton))
+
+      await waitFor(() => {
+        expect(postProfileSpy).toHaveBeenCalledWith({
+          subscriptions: {
+            marketingEmail: true,
+            marketingPush: false,
+            subscribedThemes: [],
+          },
+        })
+      })
+    })
+
+    it('should goBack and show snackbar on success', async () => {
+      mockUseAuthContext.mockReturnValueOnce(baseAuthContext)
+      mockServer.postApi<UserProfileResponse>('/v1/profile', beneficiaryUser)
+
+      render(reactQueryProviderHOC(<NotificationsSettings />))
+
+      const toggleEmail = screen.getByTestId('Interrupteur Autoriser l’envoi d’e-mails')
+      fireEvent.press(toggleEmail)
+
+      const saveButton = screen.getByText('Enregistrer')
+      await act(async () => fireEvent.press(saveButton))
+
+      expect(mockGoBack).toHaveBeenCalledTimes(1)
+      expect(mockShowSuccessSnackBar).toHaveBeenCalledWith({ message: 'Paramètre enregistré' })
+    })
+
+    it('should not go back and show snackbar on error', async () => {
```

Comment:
Plutôt `should not go back and should show`, en lisant la première fois je ne savais pas si le "not" s'appliquer à la suite de la phrase ou non